### PR TITLE
Limit action permissions

### PR DESCRIPTION
**Pull request by justinchuby**
Minimally needed permissions for actions.

Ref: https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#permission-on-pull-requests

### CODE REVIEW

**Review by AllenTiTaiWang — APPROVED** [review]
I think I was blocked by only using `pull-requests: read` last time, but never tried with this. I wonder if this affects any actions (alerts/warnings/errors). Maybe after merging we should see how it goes in the next PR?

**Comment by justinchuby** [comment]
I don't think they should since the states are determined by the github actions system and actions do not have to write any states. We will see how it goes.

**Pull request merged**
